components/xip-patch/xip-patch.sh

```diff
@@ -86,6 +86,59 @@ EOF
     fi
 }
 
+requestGardenerCerts() {
+    local subdomain
+    local shoot_domain
+    local delay
+    subdomain="kyma"
+    delay=10
+
+    echo "Getting Shoot Domain"
+    shoot_domain="$(kubectl -n kube-system get configmap shoot-info -o jsonpath='{.data.domain}')"
+    DOMAIN="${subdomain}.${shoot_domain}"
+
+    echo "Requesting certificate for domain ${DOMAIN}"
+cat <<EOF | kubectl apply -f -
+---
+apiVersion: cert.gardener.cloud/v1alpha1
+kind: Certificate
+metadata:
+  name: kyma-cert
+  namespace: kyma-installer
+spec:
+  commonName: "*.${DOMAIN}"
+EOF
+
+    while :
+    do
+    local status
+    status="$(kubectl get -n kyma-installer certificate.cert.gardener.cloud kyma-cert -o jsonpath='{.status.state}')"
+    if [ "${status}" = "Ready" ]; then
+        break
+    else
+        echo "Waiting for Certicate generation, status is ${status}"
+        sleep ${delay}
+    fi
+    done
+
+    local secret_name
+    secret_name=$(kubectl get -n kyma-installer certificate kyma-cert -o jsonpath="{.spec.secretRef.name}")
+    echo "Getting certificate from secret"
+    TLS_CERT=$(kubectl get -n kyma-installer secret  "${secret_name}" -o jsonpath="{.data['tls\.crt']}" | sed 's/ /\\ /g' | tr -d '\n')
+    TLS_KEY=$(kubectl get -n kyma-installer secret  "${secret_name}" -o jsonpath="{.data['tls\.key']}" | sed 's/ /\\ /g' | tr -d '\n')
+
+    echo "Annotating Istio Ingress Gateway with Gardener DNS"
+    kubectl -n istio-system annotate service istio-ingressgateway dns.gardener.cloud/class='garden' dns.gardener.cloud/dnsnames='*.'"${DOMAIN}"'' --overwrite
+}
+
+echo "Checking if running on Gardener"
+if [ -n "$(kubectl -n kube-system get configmap shoot-info --ignore-not-found)" ]; then
```

Comment:
check if running on gardener **and** overrides does not already exist (upgrade scenario on gardener)

Reply:
Done